### PR TITLE
planning: move path_reuse flag from PlanningContext to ReferenceLineInfo

### PR DESCRIPTION
**Pull request by jmtao**

### CODE REVIEW

File: modules/planning/common/reference_line_info.h [hunk 1]
```diff
@@ -234,6 +234,11 @@ class ReferenceLineInfo {
   void SetTurnSignal(const common::VehicleSignal::TurnSignal& turn_signal);
   void SetEmergencyLight();
 
+  void set_path_reusable(bool path_reusable) {
```
Comment (xiaoxq) [inline modules/planning/common/reference_line_info.h]: const

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc [hunk 2]
```diff
@@ -44,27 +44,31 @@ Status PathReuseDecider::Process(Frame* const frame,
...
-  bool is_change_lane_path = reference_line_info->IsChangeLanePath();
+  if (!Decider::config_.path_reuse_decider_config().reuse_path()) {
+    ADEBUG << "skipping reusing path: conf";
+    reference_line_info->set_path_reusable(false);
+    return Status::OK();
+  }
 
+  // skip path reuse if not in LANE_FOLLOW_SCENARIO
+  const ScenarioConfig_ScenarioType& scenario_type =
+      PlanningContext::Instance()->planning_status().scenario().scenario_type();
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: Make a local copy of scalar type.

File: modules/planning/common/reference_line_info.h (hunk 1 above)
Comment (jmtao) [reply modules/planning/common/reference_line_info.h]: fixed. amended. thx

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc (hunk 2 above)
Comment (jmtao) [reply modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: fixed. amended. thx

**Pull request merged**